src/finchlite/autoschedule/einsum.py

```diff
@@ -0,0 +1,262 @@
+
+        for body in plan.bodies:
+            match body:
+                case Plan(_):
+                    bodies.append(self.compile_plan(body, bindings, definitions))
+                case Query(Alias(name), Table(Literal(val), _)):
+                    bindings[name] = val
+                case Query(Alias(name), MapJoin(Literal(operation), args)):
+                    bodies.append(
+                        self.compile_mapjoin(
+                            bodies,
+                            bindings,
+                            definitions,
+                            name,
+                            body.rhs.fields,
+                            operation,
+                            args,
+                        )
+                    )
+                case Query(Alias(name), Reformat(_, MapJoin(Literal(operation), args))):
+                    bodies.append(
+                        self.compile_mapjoin(
+                            bodies,
+                            bindings,
+                            definitions,
+                            name,
+                            body.rhs.fields,
+                            operation,
+                            args,
+                        )
+                    )
+                case Query(
+                    Alias(name), Reorder(MapJoin(Literal(operation), args), idxs)
+                ):
+                    bodies.append(
+                        self.compile_mapjoin(
+                            bodies, bindings, definitions, name, idxs, operation, args
+                        )
+                    )
+                case Query(
+                    Alias(name), Aggregate(Literal(operation), Literal(init), arg, _)
+                ):
+                    bodies.append(
+                        self.compile_aggregate(
+                            bodies,
+                            bindings,
+                            definitions,
+                            name,
+                            body.rhs.fields,
+                            operation,
+                            init,
+                            arg,
+                        )
+                    )
+                case Query(
+                    Alias(name),
+                    Reformat(_, Aggregate(Literal(operation), Literal(init), arg, _)),
+                ):
+                    bodies.append(
+                        self.compile_aggregate(
+                            bodies,
+                            bindings,
+                            definitions,
+                            name,
+                            body.rhs.fields,
+                            operation,
+                            init,
+                            arg,
+                        )
+                    )
+                case Query(
+                    Alias(name),
+                    Reorder(Aggregate(Literal(operation), Literal(init), arg, _), idxs),
```

Comment:
Delete this case, you can assume the optimizer doesn't emit Reorder(Aggregate()).